pkg/controller/replica.go

```diff
@@ -138,7 +136,7 @@ func (r *ReconcileReplica) Reconcile(ctx context.Context, request reconcile.Requ
 func (r *ReconcileReplica) triggerGarbageCollection(volumeName string) {
```

Comment:
As replica `AzVolumeAttachments` for a volume gets garbage collected, disk attach capacity of nodes to which the replicas were previously attached gets freed. So, ideally, we should create failed replicas here as well.

Reply:
Good point, I was planning on creating another PR to address creating failed replicas after garbage collection to keep this one in the scope of the node_availability controller